cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -97,6 +97,26 @@ int32_t gdv_fn_populate_varlen_vector(int64_t context_ptr, int8_t* data_ptr,
   offsets[slot + 1] = offset + entry_len;
   return 0;
 }
+
+void gdv_fn_dec_from_string(const char* in, int32_t in_length,
+                            int32_t* precision_from_str, int32_t* scale_from_str,
+                            int64_t* dec_high_from_str, uint64_t* dec_low_from_str) {
+  arrow::Decimal128 dec;
+  auto status = arrow::Decimal128::FromString(std::string(in, in_length), &dec,
+                                              precision_from_str, scale_from_str);
+  *dec_high_from_str = dec.high_bits();
+  *dec_low_from_str = dec.low_bits();
+}
+
+char* gdv_fn_dec_to_string(int64_t context, int64_t x_high, uint64_t x_low,
+                           int32_t x_scale, int32_t* dec_str_len) {
+  arrow::Decimal128 dec(arrow::BasicDecimal128(x_high, x_low));
+  std::string dec_str = dec.ToString(x_scale);
+  *dec_str_len = static_cast<int32_t>(dec_str.length());
+  char* ret = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, *dec_str_len));
```

Comment:
This function can return `NULL` on failure, you need to protect against this.

Reply:
now sets error in context and returns NULLPTR